### DISCUSSION TITLE
Weird rendering (sliding number)

### DISCUSSION THREAD

**Discussion by DaniaNouro** [MuhammadSulaiman001/opengl-lab#23] [Q&A]
Hello, I am trying to add an image to each square separately.  The first square worked, but it did not work for the rest of the squares, even though I do the same steps, but the picture appears like this.  What is the cause?  Note that I follow the same method for the first square and it worked.  I tried more than one method and it did not work.  What is the solution?  or What could be the reason for this?
![Capture](https://github.com/MuhammadSulaiman001/opengl-lab/assets/142586971/8241a217-b6c4-4ddd-bf82-3272d44ba806)

I have another question
   Because I no longer understand myself
   After drawing squares, etc.
   I want to generate the initial state of the game in its random form
   How to achieve this,  I mean how to connect the logic to the drawings
   I hope that the answer is not considered a kind of cheating💔

**Comment by MuhammadSulaiman001** [comment]
Hi @DaniaNouro, For the first one, I can't tell exactly what's wrong with the weird rendering without looking at the code (which you won't post here of course).
For the second question -although it is kind of cheating-, one possible way is to create a lookup table (2d array) to reflect what is being rendered on the screen.. for each index in the array, the value would be the number of the card/texture at that position.. you should update this array whenever a card moved to the empty square, check if the values in the array represent a final state (win state) to finish the game or not!

**Comment by DaniaNouro** [comment]
استاذ لو سمحت  في حلقة ضايعة مني ماعم افهم انو مو ساويت مصفوفة ال 2d وحطيت فيها ال positions بس كيف بدو يعرف ال المربع المرسوم انو هاد موقعو السؤال  ساذج بس ماعم يزبط معي  ،+ لما يكون عندي اكتر من textures ال fragment shader بدو يستخدم الmix لكل ال textures ؟ماعم لاقي اجوبة تساعدني شوفيني استخدم ب fragment shader بهيك حالة؟😶

**Reply by MuhammadSulaiman001** [reply]
Hi @DaniaNouro, I've missed your comment (not notified)! please mention me with '@' next time.

1. We don't store the positions! as I have mentioned in my previous reply, we store "values".. so in cell[1][1], we store the value of the card in that cell (ex. [1..8]), then you update the 2d array according to keyboard strokes, finally, the updated 2d array will be rendered in `while` loop!

2. `mix` what? you can use the simplest fragment shader in this homework

```
#version 330 core
out vec4 FragColor;

in vec2 TexCoord;

// texture sampler
uniform sampler2D texture1;

void main()
{
	FragColor = texture(texture1, TexCoord);
}
```

Best.
